httpx/client.py

```diff
@@ -178,7 +178,8 @@ def merge_headers(
             )
         except HTTPError as exc:
             # Add the original request to any HTTPError
-            exc.request = request
+            if exc.request is None:
```

Comment:
This is to not replace the `request` property on a `ProxyError` which is different than the original request.

Reply:
I'd be OK with freezing that bit of information as a code comment. :-)